setup.py

```diff
@@ -21,6 +22,14 @@
 site.ENABLE_USER_SITE = "--user" in sys.argv[1:]
 
 
+def cpp_path():
+    return shlex.split(os.environ.get("CC", "cc"))[0]
+
+
+def cpp_args(args=[]):
+    return ["-E"] + shlex.split(os.environ.get("CC", ""))[1:] + args
+
```

Comment:
I rather avoid the complexity if there isn't a practical use case.
So please drop the shlex part, so something like:
```python
if os.environ.get("CC", None):
    args = ["-E",] + args
return args
```

And drop the shlex in `cpp_path`